### PR TITLE
[W5.11][W13-A2] Aadyaa Maddi

### PR DESCRIPTION
**Pull request by amad-person**
Enhancement 1: SortCommand
Added a command to display the list of persons in the AddressBook in alphabetical order of their names.

Note: Executing the sort command does not save the persons in the sorted order, it just displays them.

### CODE REVIEW

**Comment by william6364** [comment]
You could add a manual test that inserts the names in an unsorted order and then test the sort command. Currently, the names are inserted in sorted order so even if the sort command doesn't work, it would print in the correct order.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Sorts and displays all persons in the address book in alphabetical order as a list with index numbers.\n"
+            + "Example: " + COMMAND_WORD;
+
+    /** Defines ordering by name of the person in the address book. */
+    private static final Comparator<ReadOnlyPerson> PERSON_NAME_COMPARATOR = Comparator.comparing(o -> o.getName().toString());
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/SortCommand.java:22]: This line is a tad bit long; consider splitting it for better readability.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Override
+    public CommandResult execute() {
+        List<ReadOnlyPerson> sortedPersons = sortPersonListByName(addressBook.getAllPersons().immutableListView());
+
+        // TODO: save the sorted list in the addressbook, not just display
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/SortCommand.java:28]: I hope this gets done and the comment is removed ;-)

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -75,33 +77,36 @@ public Command parseCommand(String userInput) {
...
+                return prepareDelete(arguments);
 
-        case DeleteCommand.COMMAND_WORD:
-            return prepareDelete(arguments);
+            case ClearCommand.COMMAND_WORD:
+                return new ClearCommand();
 
-        case ClearCommand.COMMAND_WORD:
-            return new ClearCommand();
+            case FindCommand.COMMAND_WORD:
```
Comment (okkhoy) [inline src/seedu/addressbook/parser/Parser.java:89]: I think you messed up what was good?

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import org.junit.Test;
+
+public class SortCommandTest {
+
```
Comment (okkhoy) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:11]: empty JUnit test?

File: test/expected.txt [hunk 5]
```diff
@@ -138,6 +140,32 @@
...
+|| 	1. Aadyaa Phone: 111111 Email: this@is.not.a.real.email.com Address: 666, some street Tags: 
+|| 	2. Adam Brown Phone: 111111 Email: adam@gmail.com Address: 111, alpha street Tags: 
+|| 	3. Betsy Choo Tags: [secretive]
+|| 	4. Charlie Dickson Email: charlie.d@nus.edu.sg Address: 333, gamma street Tags: [school][friends]
+|| 	5. Dickson Ee Phone: 444444 Address: 444, delta street Tags: [friends]
+|| 	6. Esther Potato Phone: 555555 Email: esther@not.a.real.potato Tags: [tubers][starchy]
+|| 
+|| 6 persons sorted!
+|| ===================================================
+|| Enter command: || [Command entered:  list]
```
Comment (okkhoy) [inline test/expected.txt:156]: Good test case to show the effect of sort explicitly.

**Comment by okkhoy** [comment]
@amad-person 
Some comments added. Please ack & close the PR after reading comments.

**Comment by amad-person** [comment]
Acknowledged, thanks!

**Pull request closed without merging**